sqlmesh/core/plan/definition.py

```diff
@@ -441,13 +441,14 @@ def _add_restatements(self, restate_models: t.Iterable[str]) -> None:
 
             snapshots = self.context_diff.snapshots
             downstream = [
-                d
-                for d in downstream
-                if snapshots[d].is_materialized
-                and (not snapshots[d].model.disable_restatement or self.is_dev)
-                and not snapshots[d].is_seed
+                d for d in downstream if snapshots[d].is_materialized and not snapshots[d].is_seed
             ]
 
+            if not self.is_dev:
+                for d in downstream:
+                    if snapshots[d].model.disable_restatement:
+                        raise PlanError(f"Restatement is disabled for model '{d}'.")
```

Comment:
From a user perspective, do we want to raise on the first model with this condition or do we want to return all models with the condition? My personal opinion is that the latter would be more informative.